R/data.table.R

```diff
@@ -2548,18 +2548,54 @@ setnames = function(x,old,new,skip_absent=FALSE) {
   invisible(x)
 }
 
-setcolorder = function(x, neworder=key(x))
+# in anchor mode user should provide a length 1 named vector: anchor = c(before/after = anchor_col_name/index)
+setcolorder = function(x, neworder=key(x), anchor = NULL)
```

Comment:
I prefer the API discussed in the FR with two new arguments. This seems easier to read ```setcolorder(x, 1L, .before = 2L)``` than ```setcolorder(x, 1L, c(before = 2L))```.